README.md
# Introduction
This project contains a [simpleSAMLphp](http://www.simplesamlphp.org) theme 
based on the design for an identity provider login screen that is optimized 
for use on mobile devices like smart phones and tablets. It was tested on a
variety of devices.

The original design by Stroom can be found on Github Pages, and also in the 
`gh-pages` branch in this repository.
 
[Login Page](http://surfnet.github.com/simpleSAMLphp-SURFnet/index.html), 
[Error Page](http://surfnet.github.com/simpleSAMLphp-SURFnet/storing.html).

The module can be installed in simpleSAMLphp by copying it into the `modules` 
directory or by using composer.

# License
As this module takes code from simpleSAMLphp which is licensed under the LGPLv2.1
or later, this module has the same license. See included `COPYING` file.

# Installation
We assume simpleSAMLphp is installed in `/var/www/simplesamlphp`, see the
[installation instructions](http://simplesamlphp.org/docs/1.9/simplesamlphp-install). 

You can install this theme as follows:

    $ cd /var/www/simplesamlphp/modules
    $ git clone https://github.com/SURFnet/simpleSAMLphp-SURFnet.git themeSURFnet

Now you can edit the main configuration file to enable the theme, change the
following line in `/var/www/simplesamlphp/config/config.php`:

    'theme.use'             => 'default',

Into:

    'theme.use'             => 'themeSURFnet:SURFnet',

This should enable the theme. You can only see it in action when there is an
actual login screen with username and password dialog.

# Customization
If you want to use your own logo you need to modify the CSS somewhat. The logo 
included in the distribution has size 128x87. You need to modify 
`www/style.css` in the following places:

    div#header img{
            width: 128px;
            margin: 0 0 0 20px;
    }

Set `width` here to the width of the logo.

    div#header{
            position:relative;
            height: 108px;
    }

Set `height` here to the height of the logo and add 21 to it. So if your logo
has height 44, the height specified here should be `65px`.


debian/changelog
simplesamlphp-themesurfnet (20180508) unstable; urgency=medium

  * Remove unintentionally added local logo.
  * Silence spurious PHP notice in loginuserpass.php.

 -- Thijs Kinkhorst <thijs@debian.org>  Tue, 01 May 2018 10:15:55 +0000

simplesamlphp-themesurfnet (20180430) unstable; urgency=medium

  * Add template for the stepupsfo module.

 -- Thijs Kinkhorst <thijs@debian.org>  Mon, 30 Apr 2018 11:29:57 +0000

simplesamlphp-themesurfnet (20161201) unstable; urgency=medium

  * Various fixes and updates.

 -- Thijs Kinkhorst <thijs@debian.org>  Thu, 01 Dec 2016 15:18:36 +0000

simplesamlphp-themesurfnet (20140715) unstable; urgency=medium

  * Fix stray '; on error page.

 -- Thijs Kinkhorst <thijs@debian.org>  Tue, 15 Jul 2014 14:15:46 +0200

simplesamlphp-themesurfnet (20130709) unstable; urgency=low

  * Add SURFnet logo.

 -- Thijs Kinkhorst <thijs@debian.org>  Tue, 09 Jul 2013 11:55:31 +0200

simplesamlphp-themesurfnet (20130611) unstable; urgency=low

  * Initial release.

 -- Thijs Kinkhorst <thijs@debian.org>  Tue, 11 Jun 2013 13:59:05 +0200


themes/SURFnet/default/error.php
<?php
/**
 * Do not allow to frame simpleSAMLphp pages from another location.
 * This prevents clickjacking attacks in modern browsers.
 *
 * If you don't want any framing at all you can even change this to
 * 'DENY', or comment it out if you actually want to allow foreign
 * sites to put simpleSAMLphp in a frame. The latter is however
 * probably not a good security practice.
 */
header('X-Frame-Options: SAMEORIGIN');
?>
<!DOCTYPE html>

<html>

<head>
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
	
	<meta name="HandheldFriendly" content="true" />
	<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
	<meta name="robots" content="noindex, nofollow" />
	<meta name="googlebot" content="noarchive, nofollow" />
	
	<title><?php
if(array_key_exists('header', $this->data)) {
        echo $this->data['header'];
} else {
        echo 'simpleSAMLphp';
}
?></title>

	<link rel="stylesheet" type="text/css" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style.css'); ?>" />
	<link rel="stylesheet" media="screen and (max-width: 370px)" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style_320.css'); ?>" />
	<link rel="stylesheet" media="screen and (max-device-width: 480px), handheld" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style_480.css'); ?>" />

</head>

<body class="storing">
	<!-- DE WRAPPER CENTREERT DE CONTENT VAN DE WEBSITE -->
	<div id="wrapper">
	
		<!-- HEADER MET LOGO, EVENTUELE TITEL EN TAAL TOGGLE -->
		<div id="header">
			<img id="logo" src="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/logo.png'); ?>" alt="" />
			<h1 class="mainTitle"></h1>
			<ul class="langSelect">

<?php 
$includeLanguageBar = TRUE;
if (!empty($_POST)) 
	$includeLanguageBar = FALSE;
if (isset($this->data['hideLanguageBar']) && $this->data['hideLanguageBar'] === TRUE) 
	$includeLanguageBar = FALSE;

if ($includeLanguageBar) {
	$languages = $this->getLanguageList();
	$langnames = array();
	foreach($languages as $k => $v) {
		$langnames[$k] = strtoupper($k);
	}	
	$textarray = array();
	foreach ($languages AS $lang => $current) {
		$lang = strtolower($lang);
		if ($current) {
			$textarray[] = '<li class="active">' . $langnames[$lang] . "</li>";
		} else {
			$textarray[] = '<li><a href="' . htmlspecialchars(SimpleSAML_Utilities::addURLparameter(SimpleSAML_Utilities::selfURL(), array('language' => $lang))) . '">' .
				$langnames[$lang] . '</a></li>';
		}
	}
	echo join($textarray);
}
?>

<!--
				<li class="active"><a href="#">NL</a>
				<li><a href="#">EN</a>
				<li><a href="#">DE</a>
-->
			</ul>
		</div>
		<!-- EINDE HEADER MET LOGO, EVENTUELE TITEL EN TAAL TOGGLE -->
		
		<div id="content">
			<!-- CONTENT MET WITTE ACHTERGROND -->
			<div class="item">
				<h1><?php echo $this->t($this->data['dictTitle']); ?></h1>
				<p><?php
echo htmlspecialchars($this->t($this->data['dictDescr'], $this->data['parameters']));?></p>
				<!-- <p>
					Bij aanhoudende problemen kun je contact opnemen met <a href="https://servicedesk.c-college.nl" target="_blank">https://servicedesk.c-college.nl</a>, telefoon 088-4699070 
					of e&#8209;mail&nbsp;<a href="mailto: servicedesk@c-college.nl">servicedesk@c-college.nl</a>
				</p> -->
				<p>

                    <?php echo $this->t('report_trackid'); ?>
                    <?php echo $this->data['error']['trackId']; ?>
					<!-- Vermeld de volgende foutcode: <br />
					<b>1531276567327</b> -->
				</p>		
			</div>	
			<!-- EINDE CONTENT MET WITTE ACHTERGROND -->			
		</div>
		
		<!-- FOOTER -->
<!--
  		<div id="footer">
			<p><a href="#">Help</a>  <a href="#"> Privacy Policy</a>  &copy; 2012 Catharijne college</p>
		</div>
-->
		<!-- EINDE FOOTER -->
	</div>
</body>
</html>


themes/SURFnet/authX509/X509warning.php
<?php
/**
 * Do not allow to frame simpleSAMLphp pages from another location.
 * This prevents clickjacking attacks in modern browsers.
 *
 * If you don't want any framing at all you can even change this to
 * 'DENY', or comment it out if you actually want to allow foreign
 * sites to put simpleSAMLphp in a frame. The latter is however
 * probably not a good security practice.
 */
header('X-Frame-Options: SAMEORIGIN');

$warning = $this->t('{authX509:X509warning:warning}', array(
    '%days%' => htmlspecialchars($this->data['daysleft']),
));

if( $this->data['renewurl']) {
    $warning .= " " . $this->t('{authX509:X509warning:renew_url}', array(
        '%renewurl%' => $this->data['renewurl'],
    ));
} else {
    $warning .= " " . $this->t('{authX509:X509warning:renew}');
}

$this->data['header'] = $this->t('{authX509:X509warning:warning_header}');
$this->data['autofocus'] = 'proceedbutton';

?>
<!DOCTYPE html>

<html>

<head>
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
	
	<meta name="HandheldFriendly" content="true" />
	<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
	<meta name="robots" content="noindex, nofollow" />
	<meta name="googlebot" content="noarchive, nofollow" />
	
	<title><?php
if(array_key_exists('header', $this->data)) {
        echo $this->data['header'];
} else {
        echo 'simpleSAMLphp';
}
?></title>

	<link rel="stylesheet" type="text/css" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style.css'); ?>" />
	<link rel="stylesheet" media="screen and (max-width: 370px)" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style_320.css'); ?>" />
	<link rel="stylesheet" media="screen and (max-device-width: 480px), handheld" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style_480.css'); ?>" />

</head>

<body class="storing">
	<!-- DE WRAPPER CENTREERT DE CONTENT VAN DE WEBSITE -->
	<div id="wrapper">
	
		<!-- HEADER MET LOGO, EVENTUELE TITEL EN TAAL TOGGLE -->
		<div id="header">
			<img id="logo" src="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/logo.png'); ?>" alt="" />
			<h1 class="mainTitle"></h1>
			<ul class="langSelect">

<?php 
$includeLanguageBar = TRUE;
if (!empty($_POST)) 
	$includeLanguageBar = FALSE;
if (isset($this->data['hideLanguageBar']) && $this->data['hideLanguageBar'] === TRUE) 
	$includeLanguageBar = FALSE;

if ($includeLanguageBar) {
	$languages = $this->getLanguageList();
	$langnames = array();
	foreach($languages as $k => $v) {
		$langnames[$k] = strtoupper($k);
	}	
	$textarray = array();
	foreach ($languages AS $lang => $current) {
		$lang = strtolower($lang);
		if ($current) {
			$textarray[] = '<li class="active">' . $langnames[$lang] . "</li>";
		} else {
			$textarray[] = '<li><a href="' . htmlspecialchars(SimpleSAML_Utilities::addURLparameter(SimpleSAML_Utilities::selfURL(), array('language' => $lang))) . '">' .
				$langnames[$lang] . '</a></li>';
		}
	}
	echo join($textarray);
}
?>

<!--
				<li class="active"><a href="#">NL</a>
				<li><a href="#">EN</a>
				<li><a href="#">DE</a>
-->
			</ul>
		</div>
		<!-- EINDE HEADER MET LOGO, EVENTUELE TITEL EN TAAL TOGGLE -->
		
		<div id="content">
			<!-- CONTENT MET WITTE ACHTERGROND -->
			<div class="item">

<form style="display: inline; margin: 0px; padding: 0px" action="<?php echo htmlspecialchars($this->data['target']); ?>">

    <?php
        // Embed hidden fields...
        foreach ($this->data['data'] as $name => $value) {
            echo('<input type="hidden" name="' . htmlspecialchars($name) . '" value="' . htmlspecialchars($value) . '" />');
        }
    ?>
    <p><?php echo $warning; ?></p>

    <input type="submit" name="proceed" id="proceedbutton" value="<?php echo htmlspecialchars($this->t('{authX509:X509warning:proceed}')) ?>" />

</form>
			</div>	
			<!-- EINDE CONTENT MET WITTE ACHTERGROND -->			
		</div>
		
		<!-- FOOTER -->
<!--
  		<div id="footer">
			<p><a href="#">Help</a>  <a href="#"> Privacy Policy</a>  &copy; 2012 Catharijne college</p>
		</div>
-->
		<!-- EINDE FOOTER -->
	</div>
</body>
</html>


themes/SURFnet/stepupsfo/handlestatus.php
<?php
/**
 * Do not allow to frame simpleSAMLphp pages from another location.
 * This prevents clickjacking attacks in modern browsers.
 *
 * If you don't want any framing at all you can even change this to
 * 'DENY', or comment it out if you actually want to allow foreign
 * sites to put simpleSAMLphp in a frame. The latter is however
 * probably not a good security practice.
 */
header('X-Frame-Options: SAMEORIGIN');
?>

<!DOCTYPE html>

<html>

<head>
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
	
	<meta name="HandheldFriendly" content="true" />
	<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
	<meta name="robots" content="noindex, nofollow" />
	<meta name="googlebot" content="noarchive, nofollow" />
	
	<title><?php
if(array_key_exists('header', $this->data)) {
        echo $this->data['header'];
} else {
        echo 'simpleSAMLphp';
}
?></title>

	<link rel="stylesheet" type="text/css" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style.css'); ?>" />
	<link rel="stylesheet" media="screen and (max-width: 370px)" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style_320.css'); ?>" />
	<link rel="stylesheet" media="screen and (max-device-width: 480px), handheld" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style_480.css'); ?>" />

</head>

<body class="storing">
	<!-- DE WRAPPER CENTREERT DE CONTENT VAN DE WEBSITE -->
	<div id="wrapper">
	
		<!-- HEADER MET LOGO, EVENTUELE TITEL EN TAAL TOGGLE -->
		<div id="header">
			<img id="logo" src="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/logo.png'); ?>" alt="" />
			<h1 class="mainTitle"></h1>
			<ul class="langSelect">

<?php 
$includeLanguageBar = FALSE;
if (!empty($_POST)) 
	$includeLanguageBar = FALSE;
if (isset($this->data['hideLanguageBar']) && $this->data['hideLanguageBar'] === TRUE) 
	$includeLanguageBar = FALSE;

if ($includeLanguageBar) {
	$languages = $this->getLanguageList();
	$langnames = array();
	foreach($languages as $k => $v) {
		$langnames[$k] = strtoupper($k);
	}	
	$textarray = array();
	foreach ($languages AS $lang => $current) {
		$lang = strtolower($lang);
		if ($current) {
			$textarray[] = '<li class="active">' . $langnames[$lang] . "</li>";
		} else {
			$textarray[] = '<li><a href="' . htmlspecialchars(SimpleSAML_Utilities::addURLparameter(SimpleSAML_Utilities::selfURL(), array('language' => $lang))) . '">' .
				$langnames[$lang] . '</a></li>';
		}
	}
	echo join($textarray);
}
?>

<!--
				<li class="active"><a href="#">NL</a>
				<li><a href="#">EN</a>
				<li><a href="#">DE</a>
-->
			</ul>
		</div>
		<!-- EINDE HEADER MET LOGO, EVENTUELE TITEL EN TAAL TOGGLE -->
		
		<div id="content">
			<!-- CONTENT MET WITTE ACHTERGROND -->
			<div class="item">


<h1>Error while performing second factor authentication</h1>

<?php
if (
    $this->data['status'] === "urn:oasis:names:tc:SAML:2.0:status:Responder" &&
    $this->data['subStatus'] === "urn:oasis:names:tc:SAML:2.0:status:AuthnFailed" ):
?>

<p>Authentication not successful:<br/><br/>
<strong>
    <?= htmlspecialchars($this->data['statusMessage']) ?>
</strong></p>

<?php
elseif (
    $this->data['status'] === "urn:oasis:names:tc:SAML:2.0:status:Responder" &&
    $this->data['subStatus'] === "urn:oasis:names:tc:SAML:2.0:status:NoAuthnContext" ):
?>

<p>You could not be authenticated at the requested level.<br/>
<?=htmlspecialchars($this->data['statusMessage'])?></p>

<p>Do you have a token registered with the required level?<br/><br/>
Please go to the <a href="<?=$this->data['selfserviceUrl']?>">Selfservice Registration Portal</a>
to review or enroll your token.</p>

<?php
else:
?>

<p>Unexpected error occurred while performing second factor authentication.<br/><br/>
<?=htmlspecialchars($this->data['status'])?><br/>
<?=htmlspecialchars($this->data['subStatus'])?><br/>
<?=htmlspecialchars($this->data['statusMessage'])?></p>

<p>Please try again or contact your support desk.</p>

<?php
endif;
?>

			</div>	
			<!-- EINDE CONTENT MET WITTE ACHTERGROND -->			
		</div>
		
		<!-- FOOTER -->
<!--
  		<div id="footer">
			<p><a href="#">Help</a>  <a href="#"> Privacy Policy</a>  &copy; 2012 Catharijne college</p>
		</div>
-->
		<!-- EINDE FOOTER -->
	</div>
</body>
</html>


themes/SURFnet/core/loginuserpass.php
<?php
/**
 * Do not allow to frame simpleSAMLphp pages from another location.
 * This prevents clickjacking attacks in modern browsers.
 *
 * If you don't want any framing at all you can even change this to
 * 'DENY', or comment it out if you actually want to allow foreign
 * sites to put simpleSAMLphp in a frame. The latter is however
 * probably not a good security practice.
 */
header('X-Frame-Options: SAMEORIGIN');
?>
<!DOCTYPE html>

<html>

<head>
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
	
	<meta name="HandheldFriendly" content="true" />
	<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
	
	<meta name="robots" content="noindex, nofollow" />
	<title><?php
if(array_key_exists('header', $this->data)) {
        echo $this->data['header'];
} else {
        echo 'simpleSAMLphp';
}
?></title>

	<link rel="stylesheet" type="text/css" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style.css'); ?>" />
  <link rel="stylesheet" media="screen and (max-width: 370px)" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style_320.css'); ?>" />
	<link rel="stylesheet" media="screen and (max-device-width: 480px), handheld" href="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/style_480.css'); ?>" />
	
	<script type="text/javascript">
	function initiate(){
		document.getElementById('username').focus();
	}
	</script>
	
</head>

<body class="index" onload="initiate()">
	<!-- DE WRAPPER CENTREERT DE CONTENT VAN DE WEBSITE -->
	<div id="wrapper">
	
		<!-- HEADER MET LOGO, EVENTUELE TITEL EN TAAL TOGGLE -->
		<div id="header">
			<img id="logo" src="<?php echo SimpleSAML_Module::getModuleURL('themeSURFnet/logo.png'); ?>" alt="" /> <!-- DIT IS HET LOGO -->
			<h1 class="mainTitle"></h1>				        <!-- HIER KAN EEN TITEL -->
			<ul class="langSelect">

<?php 
$includeLanguageBar = FALSE;
if (!empty($_POST)) 
	$includeLanguageBar = FALSE;
if (isset($this->data['hideLanguageBar']) && $this->data['hideLanguageBar'] === TRUE) 
	$includeLanguageBar = FALSE;

if ($includeLanguageBar) {
	$languages = $this->getLanguageList();
	$langnames = array();
	foreach($languages as $k => $v) {
		$langnames[$k] = strtoupper($k);
	}	
	$textarray = array();
	foreach ($languages AS $lang => $current) {
		$lang = strtolower($lang);
		if ($current) {
			$textarray[] = '<li class="active">' . $langnames[$lang] . "</li>";
		} else {
			$textarray[] = '<li><a href="' . htmlspecialchars(SimpleSAML_Utilities::addURLparameter(SimpleSAML_Utilities::selfURL(), array('language' => $lang))) . '">' .
				$langnames[$lang] . '</a></li>';
		}
	}
	echo join($textarray);
}
?>

<!--
				<li class="active"><a href="#">NL</a>
				<li><a href="#">EN</a>
				<li><a href="#">DE</a>
-->

			</ul>
		</div>
		<!-- EINDE HEADER MET LOGO, EVENTUELE TITEL EN TAAL TOGGLE -->
		
		<div id="content">
			<!-- LOGIN VELD MET BIJBEHOREN -->
			<div class="item">
				<h1><?php echo $this->t('{login:user_pass_header}'); ?></h1>
				<p class="info"><?php echo $this->t('{login:user_pass_text}'); ?></p>

<?php
if ($this->data['errorcode'] !== NULL) {
?>

				<p class="error"><?php echo $this->t('{errors:descr_' . $this->data['errorcode'] . '}'); ?></p> <!-- ERRORHANDLER BIJ HET INLOGGEN -->

<?php
}
?>				
				<form id="login" method="POST" action="?" name="f">
					<label for="username"><?php echo $this->t('{login:username}'); ?></label> <!-- <span class="example">(bv. 123456@catherijne.nl)</span> -->
					<input type="text" name="username" id="username" value="<?php echo htmlspecialchars(@$this->data['username']); ?>" autocomplete= "off" />
					<label for="password"><?php echo $this->t('{login:password}'); ?></label>
					<input type="password" name="password" id="password" autocomplete= "off" />
					<!-- <a href="#" class="recover">Wachtwoord vergeten?</a> -->
					<input onclick="this.value='Processing...';this.disabled=true;this.form.submit();return true;" type="submit" value="<?php echo $this->t('{login:login_button}'); ?>" />

<?php
foreach ($this->data['stateparams'] as $name => $value) {
	echo('<input type="hidden" name="' . htmlspecialchars($name) . '" value="' . htmlspecialchars($value) . '" />');
}
?>

				</form>
				
			</div>
			<!-- EINDE LOGIN VELD MET BIJBEHOREN -->
			
			<!-- LIJST MET TIPS -->
			<div class="subitem">

			  <div class="createIndex">
                <h2><?php echo $this->t('{login:help_header}');?></h2>
                <p><?php echo $this->t('{login:help_text}');?></p>
<!--
       		    <ul>
					<h2>Tips:</h2>
					<li><b>Sluit uw browser af om misbruik te voorkomen.</b> Uw inlog blijft behouden voor alle websites en applicaties die ervan gebruikmaken zolang u uw browser niet afsluit.</li>
					<li>Controleer altijd voorafgaand aan het inloggen de URL van deze pagina. Deze moet beginnen met <b>https://federatie.c-college.nl</b></li>
					<li>Heeft uw aanhoudende problemen met inloggen? Neem dan contact opnemen met de servicedesk: e-mail <a href="mailto: servicedesk@c-college.nl">servicedesk@c-college.nl</a>, telefoon 088-4699070</li>
				</ul>
-->
			 </div>
			</div>

			<!-- EINDE LIJST MET TIPS -->
		</div>
		
		<!-- FOOTER -->
<!--		
        <div id="footer">
			<p><a href="#">Help</a>  <a href="#"> Privacy Policy</a>  &copy; 2012 Catharijne college</p>
		</div> 
-->
		<!-- EINDE FOOTER -->
	</div>
</body>
</html>
